src/jit/codegenxarch.cpp

```diff
@@ -757,7 +757,7 @@ void CodeGen::genCodeForBinary(GenTree* treeNode)
     }
     // now we know there are 3 different operands so attempt to use LEA
     else if (oper == GT_ADD && !varTypeIsFloating(treeNode) && !treeNode->gtOverflowEx() // LEA does not set flags
-             && (op2->isContainedIntOrIImmed() || !op2->isContained()))
+             && (op2->isContainedIntOrIImmed() || !op2->isContained()) && !treeNode->gtSetFlags())
```

Comment:
LowerAdd() is called during Lowering of IR and it replaces GT_ADD with GT_LEA.

Lower register specification is the one which is indicating whether codegen of a tree node sets/modifies ZF and SF flags.  Lower register specification is after Lowering of IR.  Hence that check cannot be used in LowerAdd.

The fact that we do GT_ADD to GT_LEA  transform early means that we might lose out on GT_RELOP optimization in this PR.